lib/logstash/inputs/lumberjack.rb

```diff
@@ -58,7 +54,6 @@ def register
     # start rejecting new connection and raise an exception
     @threadpool = Concurrent::ThreadPoolExecutor.new(
       :min_threads => 1,
-      :max_threads => @max_clients,
```

Comment:
@jordansissel this was a problem in the merge.. 
you can remove these lines https://github.com/logstash-plugins/logstash-input-lumberjack/blob/deprecate/max_clients.rb/lib/logstash/inputs/lumberjack.rb#L51-L59

Cached threadpool will allow the executor to reuse threads.